proposals/BVAL-214.md

```diff
@@ -192,3 +192,98 @@ Values passed to `validateValues()` could also be exposed through a proxy, but i
 
 * Not all beans can be proxied
 * Solution is not transparent to class-level constraint validators, they must not access fields directly, so we'd still need a vehicle for field constraints
+
+## Proposition 2
+
+This proposition is based on **Proposition 1** but changes some parts of the API.
+
+Since we depend on java 8 I think it would make sense to use `Supplier` to create the bean mocks for validation. This can look like this:
+
+    BeanValidator<ContactDataModel> contactValidator = BeanValidator.build(ContactDataModel.class);
+    contactValidator.withProperty("city", () -> cityField.getText()).
+        withProperty("zipCode", () -> zipCodeField.getText());
+
+In this example the `contactValidator` can use several times to validate the input in the UI since the values are not definied at creation of the `BeanValidator` instance but a `Supplier` is used to provide the value at runtime. By doing so the `BeanValidator` can be defined as:
+
+    public interface BeanValidator<T> {
+    
+        <V> BeanValidator<T> withProperty(String propertyName, Supplier<V> valueSupplier);
```

Comment:
So this would only allow to pass in a new value in its "current state". One of the alternatives would be to represent "actions" or "operations", e.g. allowing to represent the removal of a single element from a collection (as opposed to representing the collection itself). @emmanuelbernard had some arguments in favor of such diff-like representation, not sure though whether it makes a huge difference for actual UI validation?